config/config.yaml

```diff
@@ -108,27 +107,34 @@ storage:
     # Interval between forced flushes.
     flush_interval_sec: 5
 
-    # Max number of threads, which can be used for optimization per collection.
-    # Note: Each optimization thread will also use `max_indexing_threads` for index building.
-    # So total number of threads used for optimization will be `max_optimization_threads * max_indexing_threads`
-    # If `max_optimization_threads = 0`, optimization will be disabled.
+    # Max number of threads (jobs) for running optimizations per shard.
+    # Note: each optimization job will also use `max_indexing_threads` threads by itself for index building.
+    # If 0 - no optimization threads, optimizations will be disabled.
     max_optimization_threads: 1
 
   # Default parameters of HNSW Index. Could be overridden for each collection or named vector individually
   hnsw_index:
     # Number of edges per node in the index graph. Larger the value - more accurate the search, more space required.
     m: 16
+
     # Number of neighbours to consider during the index building. Larger the value - more accurate the search, more time required to build index.
     ef_construct: 100
+
     # Minimal size (in KiloBytes) of vectors for additional payload-based indexing.
     # If payload chunk is smaller than `full_scan_threshold_kb` additional indexing won't be used -
     # in this case full-scan search should be preferred by query planner and additional indexing is not required.
     # Note: 1Kb = 1 vector of size 256
     full_scan_threshold_kb: 10000
-    # Number of parallel threads used for background index building. If 0 - auto selection.
+
+    # Number of parallel threads used for background index building.
+    # If 0 - automatically select.
+    # Best to keep between 8 and 16 to prevent likelihood of building broken/inefficient HNSW graphs.
```

Comment:
this comment is a bit scary, we would need an official documentation to back it up

Reply:
Not related to the PR, and also I think was fixed already.

Reply:
> Best to keep between 8 and 16 to prevent likelihood of building broken/inefficient HNSW graphs.

Do you mean this specifically?

It is still in there, because a user _could_ still set something much higher. This actually happened with a user in Discord last Friday or Monday.